### PR TITLE
Add some tests that check SDP after transceiver stop

### PR DESCRIPTION
**Pull request by youennf**

### CODE REVIEW

File: webrtc/RTCRtpTransceiver-stop.html [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    answerer.setRemoteDescription(offer)
+  ]);
+  const answer = await answerer.createAnswer();
+  await Promise.all([
+    answerer.setLocalDescription(answer),
+    offerer.setRemoteDescription(answer)
+  ]);
+}
+
+promise_test(async ()=> {
```
Comment (henbos) [inline webrtc/RTCRtpTransceiver-stop.html]: nit: double indent

File: webrtc/RTCRtpTransceiver-stop.html [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    const pc1 = new RTCPeerConnection();
+
+    pc1.addTransceiver("audio", { direction: "sendonly" });
+    pc1.addTransceiver("video");
+    pc1.getTransceivers()[0].stop();
+
+    const offer = await pc1.createOffer();
+    assert_false(offer.sdp.includes("m=audio"), "offer should not contain an audio m section");
+}, "An offer should be generated even if there is no valid transceiver");
```
Comment (alvestrand) [inline webrtc/RTCRtpTransceiver-stop.html]: This description confuses me, since there is a valid video transceiver.

Comment (henbos) [inline webrtc/RTCRtpTransceiver-stop.html]: Incorrect test name

Comment (youennf) [inline webrtc/RTCRtpTransceiver-stop.html]: Title should be changed

File: webrtc/RTCRtpTransceiver-stop.html [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    pc1.addTransceiver("audio");
+
+    await negotiate(pc1, pc2);
+
+    pc1.getTransceivers()[0].direction = "sendonly";
+    pc1.getTransceivers()[0].stop();
+
+    const offer = await pc1.createOffer();
+    assert_true(offer.sdp.includes("a=sendonly"), "m section should be sendonly");
+}, "A stopped sendonly transceiver should generate an inactive m section in the offer");
```
Comment (alvestrand) [inline webrtc/RTCRtpTransceiver-stop.html]: This is testing that a stopped transceiver is present. That seems odd; you should end up with no m=audio section in the SDP.

File: webrtc/RTCRtpTransceiver-stop.html (hunk 2 above)
Comment (henbos) [reply webrtc/RTCRtpTransceiver-stop.html]: Question: does a stopped m= section not show up in the offer?

File: webrtc/RTCRtpTransceiver-stop.html (hunk 3 above)
Comment (alvestrand) [reply webrtc/RTCRtpTransceiver-stop.html]: I see now - this test does negotiation, so it's already generated the section.
This test is not testing for port=0, which is the definition of stopped, though.

**Comment by alvestrand** [comment]
Needs more review, but very good to have.

File: webrtc/RTCRtpTransceiver-stop.html [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  ]);
+  const answer = await answerer.createAnswer();
+  await Promise.all([
+    answerer.setLocalDescription(answer),
+    offerer.setRemoteDescription(answer)
+  ]);
+}
+
+promise_test(async ()=> {
+    const pc1 = new RTCPeerConnection();
```
Comment (youennf) [inline webrtc/RTCRtpTransceiver-stop.html]: add clean up

File: webrtc/RTCRtpTransceiver-stop.html (hunk 3 above)
Comment (youennf) [reply webrtc/RTCRtpTransceiver-stop.html]: Added an assert for port=0

File: webrtc/RTCRtpTransceiver-stop.html (hunk 2 above)
Comment (youennf) [reply webrtc/RTCRtpTransceiver-stop.html]: Not as per my reading of https://tools.ietf.org/html/draft-ietf-rtcweb-jsep-24#section-5.2
Firefox and Safari behaves accordingly there

**Comment by youennf** [comment]
Updated the tests according review and added some more checks.
As per my reading of https://tools.ietf.org/html/draft-ietf-rtcweb-jsep-24#section-5.2, the first test is correct.
Firefox and Safari behaves the same for all tests except the 3rd one where Firefox always set the m section as inactive.

**Review by henbos — APPROVED** [review]
LGTM

**Pull request merged**